vscode-extension/PSA_custom_Rules/BreakingChange/upcomingBreakingChange.psm1

```diff
@@ -0,0 +1,124 @@
+#Requires -Version 3.0
+
+<#
+.SYNOPSI
+    Give the tips in the upcoming breaking change places that there will be changes in the upcoming feature.
+.DESCRIPTION
+    Find all breaking change cmdlets or parameters that appear in the powershell script.
+    And give the tips that there will be changes in the feature and which breaking change there will be.
+.EXAMPLE
+    Measure-UpcomingBreakingChange -ScriptBlockAst $ScriptBlockAst
+.INPUTS
+    [System.Management.Automation.Language.ScriptBlockAst]
+.OUTPUTS
+    [OutputType([Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]])]
+.NOTES
+    None
+#>
+function Measure-UpcomingBreakingChange {
+    [CmdletBinding()]
+    [OutputType([Microsoft.Windows.Powershell.ScriptAnalyzer.Generic.DiagnosticRecord[]])]
+    Param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [System.Management.Automation.Language.ScriptBlockAst]
+        $scriptAst
+    )
+
+    Process {
+        $results = @()
+        # import functions
+        $findCmdFunctionFile = ".\PSA_custom_Rules\Find-CmdletsInFile.psm1"
+        Import-Module $findCmdFunctionFile
+        $getBreakingchangeSpecFunctionFile = ".\PSA_custom_Rules\BreakingChange\Get-BreakingChangeSpec.psm1"
+        Import-Module $getBreakingchangeSpecFunctionFile
+
+        #get the alias mapping data
+        $breakingChangePath = ".\PSA_custom_Rules\BreakingChange\BreakingchangeSpec.json"
+        $breakingchanges = Get-BreakingChangeSpec -BreakingChangePath $breakingChangePath
+
+        # get the commandAst in the file
+        $cmdletBreakingchange = Find-CmdletsInFile -RootAstNode $scriptAst
+        $typesToMessages = @{
+            "Microsoft.WindowsAzure.Commands.Common.CustomAttributes.GenericBreakingChangeAttribute"         = "The breaking change is expected to take effect from the next version.";
+            "Microsoft.WindowsAzure.Commands.Common.CustomAttributes.CmdletDeprecationAttribute"             = "The cmdlet is being deprecated. There will be no replacement for it.";
+            "Microsoft.WindowsAzure.Commands.Common.CustomAttributes.CmdletOutputBreakingChangeAttribute"    = "The output type is being deprecated without a replacement.";
+            "Microsoft.WindowsAzure.Commands.Common.CustomAttributes.CmdletParameterBreakingChangeAttribute" = "The parameter is changing."
```

Comment:
One of the biggest limitation of the current implementation is that we hard code the messages for the upcoming breaking changes. It's difficult for user to find out what needs to be done.
For example, when user runs `Resolve-Error`  there will be a warning message
```
"The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this alias to use `Resolve-AzError` instead."
```
 The action to take is clear. However in the extension they will only get an ambigous warning `"The breaking change is expected to take effect from the next version."`.

I think we need to get more data from the breaking change attribute, for example, the `ConstructorArguments` property contains the information we need:
```powershell
PS C:\Users\yeliu> $attr.ConstructorArguments | fl

ArgumentType : System.String
Value        : The `Resolve-Error` alias will be removed in a future release.  Please change any scripts that use this
               alias to use `Resolve-AzError` instead.
```
so in my opinion we need to:
- Store the `ConstructorArguments ` property in the json spec
- Display the warning message dynamically according to the arguments

Reply:
Ok, I will do it.